app/src/main/java/net/frju/flym/ui/main/MainActivity.kt

```diff
@@ -275,11 +280,33 @@ class MainActivity : AppCompatActivity(), MainNavigator, AnkoLogger {
         }
 
         AutoRefreshJobService.initAutoRefresh(this)
+
+
     }
 
+
     override fun onNewIntent(intent: Intent?) {
         super.onNewIntent(intent)
 
+        handleImplicitIntent(intent)
+    }
+
+    private fun handleImplicitIntent(intent: Intent?) {
+        // Has to be called on onStart (when the app is closed) and on onNewIntent (when the app is in the background)
+
+        //Add feed urls from Open with
+        if (intent?.action.equals(Intent.ACTION_VIEW)) {
+            var data : Uri? = intent?.data
+            FeedSearchDialog(this, data.toString()).show()
+        }
+        // Add feed urls from Share menu
+        if (intent?.action.equals(Intent.ACTION_SEND)) {
+            this.toast(intent?.hasExtra(Intent.EXTRA_TEXT).toString())
+            if (intent?.hasExtra(Intent.EXTRA_TEXT) as Boolean) {
```

Comment:
this may throw an NPE, it's better to use "if (intent?.hasExtra(Intent.EXTRA_TEXT) == true)"

Reply:
Ok